src/formats/XTC.cpp

```diff
@@ -0,0 +1,88 @@
+// Chemfiles, a modern library for chemistry file reading and writing
+// Copyright (C) Guillaume Fraux and contributors -- BSD license
+
+#include "chemfiles/formats/XTC.hpp"
+#include "chemfiles/files/XTCFile.hpp"
+
+#include "chemfiles/ErrorFmt.hpp"
+#include "chemfiles/Frame.hpp"
+using namespace chemfiles;
+
+template <> FormatInfo chemfiles::format_information<XTCFormat>() {
+    return FormatInfo("XTC").with_extension(".xtc").description("XTC binary format");
+}
+
+#define STRING_0(x) #x
+#define STRING(x) STRING_0(x)
+#define CHECK(x) check_xtc_error((x), (STRING(x)))
+
+XTCFormat::XTCFormat(std::string path, File::Mode mode, File::Compression compression)
+    : xtc_(std::move(path), mode) {
+    if (compression != File::DEFAULT) {
+        throw format_error("XTC format does not support compression");
+    }
+}
+
+size_t XTCFormat::nsteps() { return static_cast<size_t>(xtc_.get_nframes()); }
+
+void XTCFormat::read_step(size_t step, Frame& frame) {
+    step_ = static_cast<int64_t>(step);
+    CHECK(xdr_seek(xtc_, xtc_.get_offset(step_), SEEK_SET));
+    read(frame);
+}
+
+void XTCFormat::read(Frame& frame) {
+    int natoms = xtc_.get_natoms();
+    int md_step = 0;
+    float time = 0;
+    matrix box;
+    rvec* x = new rvec[natoms];
+    float precision = 0;
+
+    CHECK(read_xtc(xtc_, natoms, &md_step, &time, box, x, &precision));
+
+    frame.set("md_step", md_step); // actual step of MD Simulation
+    frame.set("time", time);       // time in pico seconds
+    frame.set("xtc_precision", precision);
+    frame.resize(static_cast<size_t>(natoms));
+
+    set_positions(x, frame);
```

Comment:
Why make these functions instead of witting the code inline?

Reply:
Thought it would be better for readability especially because there is some conversion done. That might look messy when inline. Should they be inline?

Reply:
Not necessarily, I was just wondering =)